src/common/utils.cpp

```diff
@@ -12,6 +14,41 @@ string IcebergUtils::FileToString(const string &path, FileSystem &fs) {
 	return ret_val;
 }
 
+// Function to decompress a gz file content string
+string IcebergUtils::GzFileToString(const string &path, FileSystem &fs) {
```

Comment:
Have you considered using DuckDB's [GZipFilesystem](https://github.com/duckdb/duckdb/blob/main/src/include/duckdb/common/gzip_file_system.hpp) here instead? I think that would be the preferred way to go here.

Reply:
Thank you Sam, I will try that out.

Reply:
@samansmink as per your feedback, i am using  [GzipFileSystem](https://github.com/duckdb/duckdb/blob/main/src/include/duckdb/common/gzip_file_system.hpp) to decompress the gz file and added test data and test cases on top of above feature. Please let me know in case you need more info.